lib/student.ex

```diff
@@ -16,6 +17,49 @@ defmodule Student do
     defstruct [:pid]
   end
 
+  defmodule Event do
+    use GenServer
+
+    @type t :: %__MODULE__{
+            subscribers: list(pid())
+          }
+    defstruct subscribers: []
+
+    # client side
+
+    def new() do
+      GenServer.start_link(__MODULE__, :ok)
+    end
+
+    def subscribe(student_repr, pid) do
+      {:ok, event_manager} = Student.event_manager(student_repr)
+      GenServer.call(event_manager, {:subscribe, pid})
+    end
+
+    def dispatch(student, event) do
+      GenServer.cast(student.event_manager, event)
+    end
+
+    # server side
+    @impl true
+    @spec init(any) :: {:ok, Student.Event.t()}
+    def init(_) do
+      {:ok, %Event{}}
+    end
+
+    @impl true
+    def handle_call({:subscribe, pid}, _from, %Event{subscribers: subscribers}) do
+      {:reply, :ok, %Event{subscribers: [pid | subscribers]}}
+    end
+
+    @impl true
+    def handle_cast(event, state) do
+      Enum.each(state.subscribers, fn pid -> send(pid, event) end)
+      {:noreply, state}
+    end
+  end
+
+  # client side
   @spec new(String.t()) :: :ignore | {:error, any} | {:ok, Student.Repr.t()}
```

Comment:
```suggestion
  @spec new(String.t()) :: :ignore | {:error, any()} | {:ok, Student.Repr.t()}
```
https://hexdocs.pm/elixir/typespecs.html#basic-types